packages/rollup-contracts/contracts/ExecutionManager.sol

```diff
@@ -573,24 +573,18 @@ contract ExecutionManager is FullStateManager {
      * @return True if this succeeded, false otherwise.
      */
     function createNewContract(address _newOvmContractAddress, bytes memory _ovmInitcode) internal returns (bool){
-        // Safety check the initcode -- unless the overrideSafetyChecker flag is set to true
-        if (!overrideSafetyChecker && !safetyChecker.isBytecodeSafe(_ovmInitcode)) {
-            // Contract init code is not safe.
-            return false;
-        }
         // Switch the context to be the new contract
         (address oldMsgSender, address oldActiveContract) = switchActiveContract(_newOvmContractAddress);
         // Deploy the _ovmInitcode as a code contract -- Note the init script will run in the newly set context
-        address codeContractAddress = deployCodeContract(_ovmInitcode);
-        // Get the runtime bytecode
-        bytes memory codeContractBytecode = getCodeContractBytecode(codeContractAddress);
-        // Safety check the runtime bytecode -- unless the overrideSafetyChecker flag is set to true
-        if (!overrideSafetyChecker && !safetyChecker.isBytecodeSafe(codeContractBytecode)) {
-            // Contract runtime bytecode is not safe.
+        address codeContractAddress = stateManager.deployContract(_newOvmContractAddress, _ovmInitcode);
+        // Return false if the contract failed to deploy
+        if (codeContractAddress == ZERO_ADDRESS) {
+            restoreContractContext(oldMsgSender, oldActiveContract);
             return false;
         }
         // Associate the code contract with our ovm contract
-        associateCodeContract(_newOvmContractAddress, codeContractAddress);
+        stateManager.associateCodeContract(_newOvmContractAddress, codeContractAddress);
+        bytes memory codeContractBytecode = stateManager.getCodeContractBytecode(codeContractAddress);
```

Comment:
Ahhhh sorry another non-PR-review comment--but we should be able to `getCodeContractHash` from the SM.  If it's not reflected in our tracking already, worth putting a ticket to this.

Reply:
Sorry are you talking about the partial state manager?